src/App.js

```diff
@@ -1,61 +1,161 @@
-import React from 'react';
-
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos App</h1>
-
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-          autoFocus=""
+import React, { PureComponent } from 'react';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+
+const todosFromServer = JSON.parse(localStorage.getItem('todos')) || [];
+
+class App extends PureComponent {
+  state = {
+    todoList: [...todosFromServer],
+    editingTodoId: 0,
+    activeFilter: 'all',
```

Comment:
вынесы в константы строчные значения, в таком случае тебе нужно будет если что поменять значение в одном месте, и оно поменяется везде
```
const FILTER_TYPES = {
  all: 'all',
  completed: 'completed',
  active: 'active',
}
```

И тогда можно писать так: 
```suggestion
    activeFilter: FILTER_TYPES.all,
```